### PR TITLE
editoast + api + schemas: update rolling stock model

### LINKED ISSUE #4292: update rolling stock model for BP/CC
## Implem idea

- [x] Add time to raise pantograph in the RS model
- [x] Add time for cooldown before using electricity again
- [x] Bump RJS_rolling stock_version
- [x] Update openAPI.yaml
- [x] Update RS import (probably another issue)

### PR DESCRIPTION
**Pull request by Castavo**
closes #4292

### CODE REVIEW

File: editoast/openapi.yaml [hunk 1]
```diff
@@ -2470,6 +2470,16 @@ components:
...
+        electrical_power_startup_time:
+          type: number
+          format: float
+          description: The time the train takes before actually using electrical power (in seconds)
+          example: 5.0
+        raise_pantograph_time:
+          type: number
+          format: float
+          description: The time it takes to raise this train's pantograph in s
+          example: 15.0
```
Comment (clarani) [inline editoast/openapi.yaml:2676]: Need to add the properties in the "required" list (a few lines below)

File: editoast/src/models/rolling_stock/light_rolling_stock.rs [hunk 2]
```diff
@@ -43,6 +43,8 @@ pub struct LightRollingStockModel {
     metadata: DieselJson<RollingStockMetadata>,
     power_restrictions: Option<JsonValue>,
     energy_sources: DieselJson<Vec<EnergySource>>,
+    electrical_power_startup_time: f64,
```
Comment (clarani) [inline editoast/src/models/rolling_stock/light_rolling_stock.rs]: Is this needed ? This endpoint is only used to display the list of the rolling stocks

File: front/src/common/api/osrdEditoastApi.ts [hunk 3]
```diff
@@ -1257,6 +1257,8 @@ export type RollingStockBase = {
     [key: string]: string;
   };
   energy_sources: EnergySource[];
+  electrical_power_startup_time?: number;
```
Comment (clarani) [inline front/src/common/api/osrdEditoastApi.ts]: If you add the fields to the required list, then they won't be potentially undefined in the frontend ;) 
```suggestion
  electrical_power_startup_time: number;
```

File: python/api/openapi.yaml [hunk 4]
```diff
@@ -1947,6 +1947,16 @@ components:
...
+        electrical_power_startup_time:
+          type: number
+          format: float
+          description: The time the train takes before actually using electrical power (in seconds)
+          example: 5.0
+        raise_pantograph_time:
+          type: number
+          format: float
+          description: The time it takes to raise this train's pantograph in s
+          example: 15.0
```
Comment (clarani) [inline python/api/openapi.yaml:1959]: Same here, can you add them as required ?

File: python/osrd_schemas/osrd_schemas/rolling_stock.py [hunk 5]
```diff
@@ -193,7 +194,7 @@ class RollingStock(BaseModel, extra=Extra.forbid):
     Thus power restrictions affect the effort curves, and change the power class of the rolling stock.
     """
 
-    version: Literal[RAILJSON_ROLLING_STOCK_VERSION] = Field(default=RAILJSON_ROLLING_STOCK_VERSION)
+    version: RAILJSON_ROLLING_STOCK_VERSION_TYPE = Field(default=RAILJSON_ROLLING_STOCK_VERSION)
```
Comment (clarani) [inline python/osrd_schemas/osrd_schemas/rolling_stock.py:197]: Ah nice, thanks for this

File: editoast/src/views/infra/mod.rs
Comment (clarani) [inline editoast/src/views/infra/mod.rs:1]: Nice !! Thanks

**Review by clarani — COMMENTED** [review]
Nice job, left some comments :)

File: editoast/src/models/rolling_stock/light_rolling_stock.rs (hunk 2 above)
Comment (Castavo) [reply editoast/src/models/rolling_stock/light_rolling_stock.rs]: I believe that we use `LightRollingStock` like `RollingStock` but only with simplified effort curves since they're the "heavy" part of the data.

For example we also have the mass and the rolling_resistance values which are equivalently useless for display

**Review by clarani — COMMENTED** [review]
Ready for review

**Review by clarani — APPROVED** [review]

File: editoast/src/tests/example_rolling_stock_1.json [hunk 6]
```diff
@@ -980,5 +980,7 @@
         "unit": ""
     },
     "energy_sources": [],
-    "power_restrictions": {}
+    "power_restrictions": {},
+    "electrical_power_startup_time": 5.0,
+    "raise_pantograph_time": 15.0
```
Comment (Erashin) [inline editoast/src/tests/example_rolling_stock_1.json:985]: The unit tests in views/rolling_stock.rs do depend on those .json, and are automatically verifying the whole object almost everywhere, so you are indeed updating and fixing those unit tests by adding those fields :)
However, since those fields seem to be nullable, you seem to be missing the test case where they are null. Is it possible to add this test case so as to showcase that it indeed does still work ?
I would also encourage changing the fields' values to have different values in the 3 .json. Indeed, in the update test, we take one rolling stock json and update it with the values of another. Putting different values would check that those fields are indeed updated.

File: editoast/openapi.yaml [hunk 7]
```diff
@@ -2662,6 +2662,18 @@ components:
...
+          type: number
+          nullable: true
+          format: float
+          description: The time the train takes before actually using electrical power (in seconds). Is null if the train is not electric.
+          example: 5.0
+        raise_pantograph_time:
+          type: number
+          nullable: true
+          format: float
+          description: The time it takes to raise this train's pantograph in s. Is null if the train is not electric.
```
Comment (leovalais) [inline editoast/openapi.yaml]:
```suggestion
          description: The time it takes to raise this train's pantograph in seconds. Is null if the train is not electric.
```

File: editoast/src/tests/example_rolling_stock_1.json (hunk 6 above)
Comment (leovalais) [reply editoast/src/tests/example_rolling_stock_1.json:985]: Agreed on the idea. But rather than creating some other JSON files for that specific case, I'd be more in favor of loading this JSON and changing the key value in the test case (mutating the `serde_json::Value` or the rolling stock directly) so that:
* we avoid duplicating files that we need to maintain later
* it highlights the case that's being tested while reading the code. 
Wdyt?

**Review by leovalais — APPROVED** [review]
LGTM. I've just proof-read the changes. (If it's not merged when I get back from vacation I'll test it, but I hope it will 😛)

File: editoast/src/tests/example_rolling_stock_1.json (hunk 6 above)
Comment (Erashin) [reply editoast/src/tests/example_rolling_stock_1.json:985]: Was thinking of that as well for the null test case, lgtm for sure :)

**Review by younesschrifi — APPROVED** [review]
GG Nice PR 😄

**Review by flomonster — APPROVED** [review]

**Review by eckter — APPROVED** [review]
Not tested, I just approve the core part
